services/wallet/bridge/hop.go

```diff
@@ -203,7 +203,14 @@ func (h *HopBridge) EstimateGas(fromNetwork *params.Network, toNetwork *params.N
 
 	estimation, err := ethClient.EstimateGas(ctx, msg)
 	if err != nil {
-		return 0, err
+		if !token.IsNative() {
+			// TODO: this is a temporary solution until we find a better way to estimate the gas
+			// hardcoding the estimation for other than ETH, cause we cannot get a proper estimation without having an approval placed first
+			// this is an error we're facing otherwise: `execution reverted: ERC20: transfer amount exceeds allowance`
+			estimation = 350000
```

Comment:
What s the factor u did use?

Reply:
Well, no special factor, cause I don't have any feeling about the amount, Iuri will send me some USDC on OP later and will try then, but for now I based this number on sending ETH from OP to Ethereum network and then I got `306944` for gas, then I increased it a bit and that's how I set 350k.